### DISCUSSION TITLE
Evaluating a contract using "evaluate"

### DISCUSSION THREAD

**Discussion by ayush9pandey** [pacti-org/pacti#318] [Q&A] [answered]
What is the best way to extract the numbers from the result of the “evaluate” of a contract?

I have the following:
```
print(my_contract.g.evaluate(dict_of_values))
```
Output:
```
[
  RFP <= 7294
  -RFP <= -4188
]
```

Let’s say that I want the minimum value from the inequality above (that is, 4188) as the output. I can hack my way around it by using “to_symbolic”. But I was wondering if there is a better way?

**Comment by ayush9pandey** [comment]
Also hoping to find the best way to extract the numbers from a contract assumptions and guarantees. I am using:

```
sympy.sympify(my_contract.a.to_str_list()[0]).rhs
```
to get the right hand side number in the inequality in the assumption of `my_contract`

Is there a better way?

**Comment by iincer** [comment]
Hi @ayush9pandey, have you tried using the optimization routines? One could write

```py
new_g = my_contract.g.evaluate(dict_of_values)
print(new_g.optimize(objective = [Var("RFP")], maximize = True)
```

This would be a quick way to get the max (or min) of a variable within a termlist.

**Comment by ayush9pandey** [comment, marked as answer]
I just tried this. But I think the optimize function accepts a dictionary, and not a list. So when running:

```
new_g = my_contract.g.evaluate(dict_of_values)
print(new_g.optimize(objective = {Var("RFP"): 1}, maximize = True)
```

it runs fine. Is the value that is passed the coefficient of the variable in the objective function?

**Comment by iincer** [comment]
Yes, that's right: it allows you to optimize over a linear objective. There is a convenience function called `get_variable_bounds`, but I recall it is only implemented for polyhedral contracts.
